### PR TITLE
GBE 788: 100% coverage for edit_event in a post py.test world

### PR DESCRIPTION
**Pull request by bethlakshmi**
- updated for py.text (no nose here)
- tried to adhere to same style as add_event
- refactored identical functions for add_event and edit_event testing into scheduler test functions
- found and fixed a bug - editing a class should not re-introduce the teacher if the teacher was removed.

** I just found a surprise as I got into allocate_workers - under the covers, there's a whole other part to "edit_event" - the flow that happens when this is a volunteer opportunity.  What a treat! :P  

This is a lot of code, and I'd like to get it in baseline... but let's call this "part 1", I'm writing a ticket now for part 2.

### CODE REVIEW

File: .cache/v/cache/lastfailed [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (jonkiparsky) [inline .cache/v/cache/lastfailed]: This file should be removed from the PR

File: expo/.cache/v/cache/lastfailed [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{
```
Comment (jonkiparsky) [inline expo/.cache/v/cache/lastfailed]: Likewise

File: expo/tests/scheduler/test_edit_event.py [hunk 3]
```diff
@@ -1,97 +1,329 @@
...
+            data=form_data,
+            follow=True)
+
+        self.assert_good_post(response,
+                              self.context,
+                              self.context.days[0],
+                              self.context.room)
+        teachers = self.context.sched_event.get_direct_workers('Teacher')
+        self.assertEqual(len(teachers), 0)
+        self.assertIn('<select id="id_event-teacher" name="event-teacher">\n' +
```
Comment (jonkiparsky) [inline expo/tests/scheduler/test_edit_event.py]: FWIW, python concatenates adjacent strings, so the + here is not necessary. Not a blocker, but worth knowing.

File: expo/scheduler/views.py [hunk 4]
```diff
@@ -795,24 +825,8 @@ def add_event(request, eventitem_id, event_type='Class'):
...
-                s_event.unallocate_role('Panelist')
-                for panelist in data['panelists']:
-                    s_event.allocate_worker(panelist.workeritem, 'Panelist')
-
-            if data['staff_lead']:
-                s_event.unallocate_role('Staff Lead')
-                s_event.allocate_worker(data['staff_lead'].workeritem,
-                                        'Staff Lead')
-
+            set_single_role(s_event, data)
```
Comment (jonkiparsky) [inline expo/scheduler/views.py]: Nice, love it.

File: expo/tests/scheduler/test_edit_event.py [hunk 5]
```diff
@@ -1,97 +1,329 @@
...
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        form_data = get_sched_event_form(self.context)
+        form_data['event-location'] = 'bad room'
+        response = self.client.post(
+            url,
+            data=form_data)
 
         self.assertEqual(response.status_code, 200)
```
Comment (jonkiparsky) [inline expo/tests/scheduler/test_edit_event.py]: Hm. This looks a lot like the assert_good_post code, except the values are just a little different.

**Comment by jonkiparsky** [comment]
Looks okay. If you can get rid of those two data files, I'll accept it.

**Comment by jonkiparsky** [comment]
Oh, hang on:

```
expo/scheduler/views.py:775:1: E302 expected 2 blank lines, found 1
expo/scheduler/views.py:778:17: E128 continuation line under-indented for visual indent
expo/scheduler/views.py:779:17: E128 continuation line under-indented for visual indent
expo/scheduler/views.py:786:1: E302 expected 2 blank lines, found 1
expo/scheduler/views.py:793:18: E111 indentation is not a multiple of four
expo/scheduler/views.py:795:1: W293 blank line contains whitespace
expo/tests/contexts/class_context.py:54:35: E128 continuation line under-indented for visual indent
expo/tests/contexts/class_context.py:54:42: E231 missing whitespace after ','
expo/tests/contexts/class_context.py:54:44: E231 missing whitespace after ','
expo/tests/functions/scheduler_functions.py:8:1: E302 expected 2 blank lines, found 1
expo/tests/functions/scheduler_functions.py:24:1: E302 expected 2 blank lines, found 1
expo/tests/functions/scheduler_functions.py:29:13: E122 continuation line missing indentation or outdented
expo/tests/functions/scheduler_functions.py:33:1: E302 expected 2 blank lines, found 1
expo/tests/functions/scheduler_functions.py:43:1: E302 expected 2 blank lines, found 1
expo/tests/functions/scheduler_functions.py:48:23: E127 continuation line over-indented for visual indent
expo/tests/scheduler/test_edit_event.py:233:80: E501 line too long (80 > 79 characters)

```

File: expo/.cache/v/cache/lastfailed (hunk 2 above)
Comment (bethlakshmi) [inline expo/.cache/v/cache/lastfailed]: is there a way to exclude this from git?  I've been manually removing it every checkin from the check list, but second I miss one... - I tried adding the path to git ignore, but it doesn't work.  Is there a place to configure where this cache shows up?  It started appearing with py.test

**Comment by bethlakshmi** [comment]
OK @jonkiparsky .  Pep8 and cache cleanup done.

Sorry about the pep8!  Thought I'd done it!

**Pull request merged**